aineko/extras/connectors/wss.py

```diff
@@ -0,0 +1,209 @@
+                if dataset != "logging":
+                    producer.produce(message)
+            self.retry_count = 0
+        except json.decoder.JSONDecodeError as err:
+            if self.retry_count < self.wss_params.max_retries:
+                self.retry_count += 1
+                self.log(
+                    f"Failed to parse message: {str(raw_message)}. "
+                    f"The following error occured: {str(err)} "
+                    f"Reconnecting in {str(self.wss_params.retry_sleep)} "
+                    "seconds...",
+                    level="error",
+                )
+                time.sleep(self.wss_params.retry_sleep)
+                self.create_subscription()
+            else:
+                raise ValueError(
+                    "Retry count exceeded max retries "
+                    f"({str(self.wss_params.max_retries)}). "
+                    f"Failed to parse message: {str(raw_message)}. "
+                    f"The following error occured: {str(err)}"
+                ) from err
+
+    def create_subscription(self) -> None:
+        """Creates a subscription on the websocket."""
+        try:
+            self.log(f"Creating subscription to {str(self.wss_params.url)}...")
+            self.ws.connect(
+                url=self.wss_params.url, header=self.wss_params.header
+            )  # type: ignore
+
+            if self.wss_params.init_messages:
+                # Send initialization messages
+                for init_msg in self.wss_params.init_messages:
+                    self.ws.send(json.dumps(init_msg))
+                    message = self.ws.recv()
+                    self.log(
+                        f"Sent initialization message to "
+                        f"{str(self.wss_params.url)}. "
+                        f"Acknowledged initialization message: {str(message)}"
+                    )
+
+            ack_message = self.ws.recv()
+            self.log(
+                f"Subscription to {str(self.wss_params.url)} created. "
+                f"Acknowledged subscription message: {str(ack_message)}"
+            )
+
+            self.retry_count = 0
+        except Exception as err:  # pylint: disable=broad-except
+            if self.retry_count < self.wss_params.max_retries:
+                self.log(
+                    "Encountered error when attempting to connect to "
+                    f"{str(self.wss_params.url)}. Will retry in "
+                    f"{str(self.wss_params.retry_sleep)} seconds"
+                )
+                self.retry_count += 1
+                time.sleep(self.wss_params.retry_sleep)
+                self.create_subscription()
+            else:
+                raise ValueError(
+                    "Retry count exceeded max retries. "
+                    "Failed to create subscription to "
+                    f"{str(self.wss_params.url)}. "
+                    f"The following error occured: {str(err)}"
```

Comment:
```suggestion
                    f"The following error occurred: {err}"
```
typo + redundant type casting

Reply:
fixed here: https://github.com/aineko-dev/aineko/pull/121/commits/d71cd6b4c6b3f85bd4230ef93074e42ece1fdf78